src/RageSoundReader_Chain.cpp

```diff
@@ -186,19 +186,19 @@ void RageSoundReader_Chain::Finish()
 	m_iActualSampleRate = GetSampleRateInternal();
 	if( m_iActualSampleRate == -1 )
 	{
-		for (RageSoundReader *it : m_apLoadedSounds)
+		for (auto &pSound : m_apLoadedSounds)
 		{
-			RageSoundReader_Resample_Good *pResample = new RageSoundReader_Resample_Good( it, m_iPreferredSampleRate );
-			it = pResample;

```

Comment:
> Maybe someone can shed some more light on what's happening here.

The issue is the `it = pResample` is only modifying the local copy of it (rangefor loops are copy by value unless using a reference), so the pointer in the original vector wasn't ever actually being updated.

Your change makes it so pSound actually *is* the pointer in the vector so it's updated properly.